src/IO/ReadHelpers.cpp

```diff
@@ -1111,4 +1111,12 @@ bool loadAtPosition(ReadBuffer & in, DB::Memory<> & memory, char * & current)
     return loaded_more;
 }
 
+bool isStringNumber(StringRef str)
+{
+    for (size_t i = 0; i != str.size; ++i)
+        if (!isdigit(str.data[i]))
```

Comment:
This function depends on locale that may give surprising results.
Let's replace to `isNumericASCII` from `StringUtils`.